bpf_tests: Loading of unused programs causes test failures on older kernels

Running the BPF unit tests on older (pre 5.8) kernels results in errors while loading.

```
ubuntu@dev:~/go/src/github.com/cilium/cilium$ make -C test run_bpf_tests
make: Entering directory '/home/ubuntu/go/src/github.com/cilium/cilium/test'
make -C ../bpf/tests all
make[1]: Entering directory '/home/ubuntu/go/src/github.com/cilium/cilium/bpf/tests'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/ubuntu/go/src/github.com/cilium/cilium/bpf/tests'
go test ./bpf_tests -exec sudo -bpf-test-path /home/ubuntu/go/src/github.com/cilium/cilium/bpf/tests 
--- FAIL: TestBPF (0.18s)
    bpf_test.go:215: Unknown message type=0 arg1=3520 arg2=65664
    bpf_test.go:215: Conntrack lookup 1/2: src=192.168.0.1:22334 dst=172.16.10.1:80
    bpf_test.go:215: Conntrack lookup 2/2: nexthdr=132 flags=4
    bpf_test.go:215: CT verdict: New, revnat=0
    bpf_test.go:215: Service backend slot lookup: slot=1, dport=80
    bpf_test.go:215: Conntrack create: proxy-port=0 revnat=256 src-identity=0 lb=0.0.0.0
    bpf_test.go:215: Loopback SNAT from=192.168.0.1 to=10.245.255.31
    bpf_test.go:215: Conntrack lookup 1/2: src=10.245.255.31:22334 dst=192.168.0.1:80
    bpf_test.go:215: Conntrack lookup 2/2: nexthdr=132 flags=1
    bpf_test.go:215: CT verdict: New, revnat=0
    bpf_test.go:215: Failed to map addr=192.168.0.1 to identity
    bpf_test.go:215: Conntrack create: proxy-port=0 revnat=256 src-identity=1048575 lb=10.245.255.31
    bpf_test.go:215: Attempting local delivery for container id 0 from seclabel 1048575
    bpf_test.go:215: Conntrack lookup 1/2: src=192.168.0.1:80 dst=10.245.255.31:22334
    bpf_test.go:215: Conntrack lookup 2/2: nexthdr=132 flags=1
    bpf_test.go:215: CT entry found lifetime=2741446, revnat=256
    bpf_test.go:215: CT verdict: Reply, revnat=256
    bpf_test.go:215: Failed to map addr=10.245.255.31 to identity
    bpf_test.go:215: Reverse NAT lookup, index=256
    bpf_test.go:215: Performing reverse NAT, address=172.16.10.1 port=80
    bpf_test.go:215: Loopback reverse SNAT from=10.245.255.31 to=192.168.0.1
    bpf_test.go:215: Attempting local delivery for container id 0 from seclabel 1048575
    bpf_test.go:154: new coll: program cil_sock6_getpeername: load program: invalid argument
FAIL
FAIL	github.com/cilium/cilium/test/bpf_tests	0.225s
FAIL
make: *** [Makefile:88: run_bpf_tests] Error 1
make: Leaving directory '/home/ubuntu/go/src/github.com/cilium/cilium/test'
```

Root cause is that the `cil_sock6_getpeername` program is a `cgroup/getpeername6` program which was introduced in kernel 5.8.
Additionally, cgroups have no support for BPF_PROG_RUN, so loading such a program is useless in the first place. Therefor my suggested fix for this issue is to modify the runner so it only loads programs that are relavent(executed as part of the test suite or are values in a tail call map).

--- a/test/bpf_tests/bpf_test.go
+++ b/test/bpf_tests/bpf_test.go
@@ -27,6 +27,7 @@ import (
 	"github.com/cilium/coverbee"
 	"github.com/cilium/ebpf"
 	"github.com/cilium/ebpf/perf"
+	"github.com/cilium/ebpf/rlimit"
 	"github.com/davecgh/go-spew/spew"
 	"github.com/golang/protobuf/proto"
 	"github.com/vishvananda/netlink/nl"
@@ -51,6 +52,10 @@ var (
 func TestBPF(t *testing.T) {
 	if testPath == nil || *testPath == "" {
 		t.Skip("Set -bpf-test-path to run BPF tests")
+	}
+
+	if err := rlimit.RemoveMemlock(); err != nil {
+		t.Log(err)
 	}
 
 	entries, err := os.ReadDir(*testPath)
@@ -269,9 +274,21 @@ func loadAndPrepSpec(t *testing.T, elfPath string) *ebpf.CollectionSpec {
 		mspec.Pinning = ebpf.PinNone
 	}
 
-	// Detect program type mismatches
 	var progTestType ebpf.ProgramType
-	for _, prog := range spec.Programs {
+	for progName, prog := range spec.Programs {
+		switch prog.Type {
+		case ebpf.XDP, ebpf.SchedACT, ebpf.SchedCLS:
+		case ebpf.UnspecifiedProgram:
+		default:
+			t.Logf(
+				"program '%s' has program type '%s' which doesn't have BPF_PROG_RUN support, not loading it",
+				prog.Name,
+				prog.Type,
+			)
+			delete(spec.Programs, progName)
+			continue
+		}
+
 		if progTestType != prog.Type {
 			if progTestType == ebpf.UnspecifiedProgram {
 				progTestType = prog.Type

